vector-im/element-android
# Delete unused client information from account data (PSG-871)

## Type of change


## Content

With new session manager implementation we started to store information of user devices in account data. If user logouts / delete a session we need to delete the corresponding client information to keep the account data as clean as possible. Backend has now a new endpoint to be able to delete account data events. This PR implements this feature.

## Motivation and context
Implement [MSC-3391](https://github.com/ShadowJonathan/matrix-doc/blob/account-data-delete/proposals/3391-account-data-delete.md#sync)



## Screenshots / GIFs





## Tests



- Enable client info recording from Labs settings
- Login from another client
- Debug or use Postman to see the new session is recorded: `account_data/io.element.matrix_client_information.DEVICE`
- Logout from this client
- Check account data to see this record is deleted from account data

## Tested devices

- OS version(s):

## Checklist
